File: src/main/java/com/lmptech/intune/controllers/CommonController.java
package com.lmptech.intune.controllers;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController()
@RequestMapping("common")
public class CommonController {

    @GetMapping("ping")
    public String ping() {
        return "pong";
    }
}

File: src/main/kotlin/com/x64tech/intune/controllers/CommonController.kt
package com.x64tech.intune.controllers

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/common")
class CommonController () {
    @GetMapping("/public")
    fun publicRoute(): ResponseEntity<String> {
        return ResponseEntity.ok().body("OK PUBLIC ROUTE")
    }

    @GetMapping("/private")
    fun privateRoute(): ResponseEntity<String> {
        return ResponseEntity.ok().body("OK PRIVATE ROUTE")
    }
}